client/human/HumanClientApp.cpp

```diff
@@ -220,13 +220,23 @@ HumanClientApp::HumanClientApp(int width, int height, bool calculate_fps, const
 
     SetMinDragTime(0);
 
-    m_ui = boost::shared_ptr<ClientUI>(new ClientUI());
-
-    if ((GetOptionsDB().Get<bool>("UI.sound.music-enabled")))
-        Sound::GetSound().PlayMusic(GetOptionsDB().Get<std::string>("UI.sound.bg-music"), -1);
+    bool inform_user_sound_failed(false);
+    try {
+        if (GetOptionsDB().Get<bool>("UI.sound.enabled") || GetOptionsDB().Get<bool>("UI.sound.music-enabled"))
+            Sound::GetSound().Enable();
+
+        if ((GetOptionsDB().Get<bool>("UI.sound.music-enabled")))
+            Sound::GetSound().PlayMusic(GetOptionsDB().Get<std::string>("UI.sound.bg-music"), -1);
+
+        Sound::GetSound().SetMusicVolume(GetOptionsDB().Get<int>("UI.sound.music-volume"));
+        Sound::GetSound().SetUISoundsVolume(GetOptionsDB().Get<int>("UI.sound.volume"));
+    } catch (std::runtime_error const & e) {
+        if (std::string(e.what()) != "ERROR_SOUND_INITIALIZATION_FAILED")
```

Comment:
if not defining a new exception class, at least put this string into a constant somewhere, rather than retyping it in each place it's checked...